From 27db7ba15189cb166b776e4e1ebeab0fdb5a1538 Mon Sep 17 00:00:00 2001
From: rohithsharmaks <rohithsharmaks@apache.org>
Date: Sun, 24 Jan 2016 23:36:15 +0530
Subject: [PATCH 1278/2848] YARN-4613. Fix test failure in
 TestClientRMService#testGetClusterNodes. (Takashi
 Ohnishi via rohithsharmaks)

(cherry picked from commit 10dc2c049304671f0ed9fd737a30cd843427b53e)

Change-Id: I29af36c6a5831177e0e1b533ae63069b710008cf
---
 .../resourcemanager/TestClientRMService.java       |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
index 15aca42..120b334 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/TestClientRMService.java
@@ -237,6 +237,7 @@ protected ClientRMService createClientRMService() {
 
     // Now make the node unhealthy.
     node.nodeHeartbeat(false);
+    rm.NMwaitForState(node.getNodeId(), NodeState.UNHEALTHY);
 
     // Call again
     nodeReports = client.getClusterNodes(request).getNodeReports();
-- 
1.7.9.5

